x/attribute/keeper/keeper.go

```diff
@@ -438,3 +500,58 @@ func (k Keeper) PopulateAddressAttributeNameTable(ctx sdk.Context) {
 		k.IncAttrNameAddressLookup(ctx, attr.Name, attr.GetAddressBytes())
 	}
 }
+
+// DeleteExpiredAttributes find and delete expired attributes returns the total deleted
+// limit sets the max amount to delete in a call, 0 for not limit
+func (k Keeper) DeleteExpiredAttributes(ctx sdk.Context, limit int) int {
+	store := ctx.KVStore(k.storeKey)
+
+	iterator := store.Iterator(types.AttributeExpirationKeyPrefix, types.GetAttributeExpireTimePrefix(ctx.BlockTime()))
+	defer iterator.Close()
+	count := 0
+	for ; iterator.Valid(); iterator.Next() {
+		attrKey := types.GetAddrAttributeKeyFromExpireKey(iterator.Key())
+		bz := store.Get(attrKey)
+		if bz != nil {
+			var attribute types.Attribute
+			if err := k.cdc.Unmarshal(bz, &attribute); err == nil {
+				// delete attribute from store
+				store.Delete(attrKey)
+				// dec name to address lookup table count
+				k.DecAttrNameAddressLookup(ctx, attribute.Name, attribute.GetAddressBytes())
+
+				deleteExpirationEvent := types.NewEventAttributeExpiredDelete(attribute)
+				if err = ctx.EventManager().EmitTypedEvent(deleteExpirationEvent); err != nil {
+					ctx.Logger().Error(fmt.Sprintf("failed to emit typed event %v", err))
+				}
+				count++
+			} else {
+				ctx.Logger().Error(fmt.Sprintf("unable to unmarshal attribute to delete key: %v error: %v", attrKey, err))
```

Comment:
we might want to delete the index record in this case ... because as written this will fail with this error on every block going forward...

Reply:
Thanks, good catch.